### ISSUE TITLE
problem with using terraform template_file

### ISSUE THREAD

**Issue by dhrp** [Invicton-Labs/terraform-null-assertion#2]
It looks like your last commit contains a number of instances of use of data "template_file" which is deprecated and removed from hashicorp. 

Your use here:
https://github.com/Invicton-Labs/terraform-null-assertion/blob/main/main.tf#L9

LInks to the removal notice:
https://registry.terraform.io/providers/hashicorp/template/latest/docs/data-sources/file
https://github.com/hashicorp/terraform-provider-template/issues/85

I'm reporting this because my templates stopped working with the following error:

```
Error: Incompatible provider version
│ 
│ Provider registry.terraform.io/hashicorp/template v2.2.0 does not have a package
│ available for your current platform, darwin_arm64.
│ 
```

This library appears to be a deep dependency.

The issue will show on Mac M1 (or newer) machines.

**Comment by KyleKotowick** [comment]
Thank you for reporting this. The need for using a data source this way is documented in the release notes and in the file you linked to, but I'll need to find another that works on MacOS.

**Comment by KyleKotowick** [comment]
@dhrp This should be fixed in the new release v0.2.4. I've also added automated tests that ensure that the module works on all operating systems and Terraform versions.

Feel free to create a new issue if there are any remaining problems.
